engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -271,20 +252,16 @@ private void preRenderUpdate(RenderingStage renderingStage) {
         resetStats();
 
         currentRenderingStage = renderingStage;
-        if (currentRenderingStage == RenderingStage.MONO || currentRenderingStage == RenderingStage.LEFT_EYE) {
-            isFirstRenderingStageForCurrentFrame = true;
-        } else {
-            isFirstRenderingStageForCurrentFrame = false;
-        }
+        isFirstRenderingStageForCurrentFrame = currentRenderingStage == RenderingStage.MONO || currentRenderingStage == RenderingStage.LEFT_EYE;
 
         // this is done to execute this code block only once per frame
         // instead of once per eye in a stereo setup.
         if (isFirstRenderingStageForCurrentFrame) {
             timeSmoothedMainLightIntensity = TeraMath.lerp(timeSmoothedMainLightIntensity, getMainLightIntensityAt(playerCamera.getPosition()), secondsSinceLastFrame);
 
             playerCamera.update(secondsSinceLastFrame);
-            positionShadowMapCamera();
-            shadowMapCamera.update(secondsSinceLastFrame);
+
+            shadowMap.preRender(secondsSinceLastFrame);
```

Comment:
Throughout Terasology's code the methods update() and render() are common occurrences.  In particular it is known that updates happen before the rendering and might occur multiple times per frame. I'd recommend in this case to stick to that and have shadowMapNode.update(secondsSinceLastFrame) instead.